source/components/inputs/dateTime/dateTime.ng1.ts

```diff
@@ -117,6 +117,7 @@ export class DateTimeController extends InputController {
 		});
 
 		this.ngModel.$parsers.push((value: string): moment.Moment => {
+			this.timezone = __timezone.timezoneService.currentTimezone;
```

Comment:
I feel like you only want to do this if the timezone is currently null. In 'most' cases, this should get set in the formatters long before this